src/crypto/PSASpake2p.cpp

```diff
@@ -179,22 +180,30 @@ CHIP_ERROR PSASpake2p_P256_SHA256_HKDF_HMAC::KeyConfirm(const uint8_t * in, size
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR PSASpake2p_P256_SHA256_HKDF_HMAC::GetKeys(uint8_t * out, size_t * out_len)
+CHIP_ERROR PSASpake2p_P256_SHA256_HKDF_HMAC::GetKeys(SessionKeystore & keystore, HkdfKeyHandle & key)
 {
-    VerifyOrReturnError(out != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(out_len != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-
     /*
-     * TODO: either:
-     * - use psa_pake_shared_secret() proposed in https://github.com/ARM-software/psa-api/issues/86
-     * - refactor Matter's GetKeys API to take an abstract shared secret instead of raw secret bytes.
+     * TODO: use psa_pake_shared_secret() proposed in https://github.com/ARM-software/psa-api/issues/86
      */
-    oberon_spake2p_operation_t & oberonCtx = mOperation.MBEDTLS_PRIVATE(ctx).oberon_pake_ctx.ctx.oberon_spake2p_ctx;
 
-    VerifyOrReturnError((oberonCtx.hash_len / 2) <= *out_len, CHIP_ERROR_BUFFER_TOO_SMALL);
+    psa_key_derivation_operation_t * kdf = Platform::New<psa_key_derivation_operation_t>();
+    Platform::UniquePtr<psa_key_derivation_operation_t> kdfPtr(kdf);
+
+    VerifyOrReturnError(kdfPtr, CHIP_ERROR_NO_MEMORY);
+
+    *kdfPtr = PSA_KEY_DERIVATION_OPERATION_INIT;
+
+    psa_status_t status = psa_key_derivation_setup(kdfPtr.get(), PSA_ALG_HKDF(PSA_ALG_SHA_256));
+    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+
+    status = psa_pake_get_implicit_key(&mOperation, kdfPtr.get());
+    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+
+    auto & hkdfKeyHandle           = key.AsMutable<PsaHkdfKeyHandle>();
+    hkdfKeyHandle.mKeyDerivationOp = kdfPtr.get();
+    hkdfKeyHandle.mIsKeyId         = false;
 
-    memcpy(out, oberonCtx.shared, oberonCtx.hash_len / 2);
-    *out_len = oberonCtx.hash_len / 2;
+    kdfPtr.release();
```

Comment:
I guess it's fine and more clean.. but do we need to release it explicitly?

Reply:
Could you clarify? We have to release it to inform the UniquePtr that the memory will be deleted in some other place (by the `DestroyKey` method). If we won't do that, the UniquePtr will delete key just after we leave the scope.